src/main/java/org/spongepowered/common/data/processor/data/entity/TameableDataProcessor.java

```diff
@@ -0,0 +1,163 @@
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.spongepowered.common.data.util.DataUtil.getData;
+
+import com.google.common.base.Optional;
+import com.mojang.util.UUIDTypeAdapter;
+import net.minecraft.entity.passive.EntityTameable;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataTransactionBuilder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.DataView;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableTameableData;
+import org.spongepowered.api.data.manipulator.mutable.entity.TameableData;
+import org.spongepowered.api.data.merge.MergeFunction;
+import org.spongepowered.api.data.value.BaseValue;
+import org.spongepowered.api.data.value.immutable.ImmutableValue;
+import org.spongepowered.api.data.value.mutable.OptionalValue;
+import org.spongepowered.api.service.persistence.InvalidDataException;
+import org.spongepowered.common.data.DataProcessor;
+import org.spongepowered.common.data.manipulator.immutable.entity.ImmutableSpongeTameableData;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeTameableData;
+import org.spongepowered.common.data.processor.common.AbstractSpongeDataProcessor;
+import org.spongepowered.common.data.value.AbstractBaseValue;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeOptionalValue;
+
+import javax.annotation.Nullable;
+import java.util.UUID;
+
+public class TameableDataProcessor extends AbstractSpongeDataProcessor<TameableData, ImmutableTameableData> {
+
+    @Override
+    public boolean supports(DataHolder dataHolder) {
+        return dataHolder instanceof EntityTameable;
+    }
+
+    @Override
+    public Optional<TameableData> from(DataHolder dataHolder) {
+        if(dataHolder instanceof EntityTameable) {
+            final Optional<UUID> uuidOptional = getTamer(((EntityTameable) dataHolder));
+            return Optional.<TameableData>of(new SpongeTameableData(uuidOptional.orNull()));
+        } else {
+            return Optional.absent();
+        }
+    }
+
+    @Override
+    public Optional<TameableData> fill(DataHolder dataHolder, TameableData manipulator, MergeFunction overlap) {
+        if(dataHolder instanceof EntityTameable) {
+            final TameableData merged = checkNotNull(overlap, "MergeFunction cannot be null!").merge(checkNotNull(manipulator).copy(), from(dataHolder).orNull());
+            manipulator.set(Keys.TAMED_OWNER, merged.owner().get());
+            return Optional.of(manipulator);
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<TameableData> fill(final DataContainer container, final TameableData tameableData) {
+        tameableData.set(Keys.TAMED_OWNER, getData(container, Keys.TAMED_OWNER));
```

Comment:
I'm pretty sure that `MemoryDataView` doesn't know how to deserialize a `UUID` from string, so you'd have to explicitly deserialize.